src/backend/mod.rs

```diff
@@ -2,31 +2,43 @@ mod alacritty;
 mod kitty;
 mod urxvt;
 use super::config::Config;
+use crate::config::Backend;
 use crate::error::GlrnvimError;
 use std::path::PathBuf;
 
-pub const BACKEND_LIST: &'static [&'static str] = &[
-    alacritty::ALACRITTY_NAME,
-    urxvt::URXVT_NAME,
-    kitty::KITTY_NAME,
-];
-
 pub trait Functions {
     fn create_command(&mut self, config: &Config) -> std::process::Command;
 }
 
-pub fn init(backend_name: &str) -> Result<Box<dyn Functions>, GlrnvimError> {
-    let name = backend_name.to_lowercase();
+pub fn init(config: &Config) -> Result<Box<dyn Functions>, GlrnvimError> {
```

Comment:
I do not like this function, have any ideas how this can be improved?

Reply:
what about delay the `exe_path`/`find_executable` call until `create_command` starts?